src/Impostor.Server/Net/State/Game.Api.cs

```diff
@@ -20,24 +23,45 @@ public void BanIp(IPAddress ipAddress)
             _bannedIps.Add(ipAddress);
         }
 
-        // TODO This no longer does anything, it was replaced by LogicOptions
         public async ValueTask SyncSettingsAsync()
         {
             if (Host?.Character == null)
             {
-                throw new ImpostorException("Attempted to set infected when the host was not spawned.");
+                throw new ImpostorException("Attempted to change settings when the host was not spawned.");
             }
 
-            using (var writer = StartRpc(Host.Character.NetId, RpcCalls.SyncSettings))
+            var gameManager = FindObjectByType<InnerGameManager>();
+            if (gameManager == null)
             {
-                // Someone will probably forget to do this, so we include it here.
-                // If this is not done, the host will overwrite changes later with the defaults.
-                Options.IsDefaults = false;
-
-                Rpc02SyncSettings.Serialize(writer, Options);
+                throw new ImpostorException("Attempted to change options when the game manager was not spawned.");
+            }
 
-                await FinishRpcAsync(writer);
+            var gameOptionsTag = gameManager.GetGameLogicTag(gameManager.LogicOptions);
+            if (gameOptionsTag == null)
+            {
+                throw new ImpostorException("Attempted to change options when the LogicOptions was not spawned.");
             }
+
+            // Someone will probably forget to do this, so we include it here.
+            // If this is not done, the host will overwrite changes later with the defaults.
+            Options.IsDefaults = false;
+
+            using var writer = MessageWriter.Get(MessageType.Reliable);
+
+            writer.StartMessage(MessageFlags.GameData);
+            Code.Serialize(writer);
+
+            writer.StartMessage(GameDataTag.DataFlag);
+            writer.WritePacked(gameManager.NetId);
+
+            writer.StartMessage((byte)gameOptionsTag);
+            GameOptionsFactory.Serialize(writer, Options);
```

Comment:
Use `LogicOptions.Serialize` here, somehow?

Reply:
It will only replace the `GameOptionsFactory.Serialize(writer, Options);` line then. But we don't use the `LogicOptions.Serialize` call anywhere else as said by an earlier comment.